presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/CassandraSplitManager.java

```diff
@@ -77,7 +92,20 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
             }
         }
 
-        return new FixedSplitSource(getSplitsForPartitions(cassandraTableHandle, partitions, cassandraTableHandle.getClusteringKeyPredicates()));
+        return new FixedSplitSource(getSplitsForPartitions(cassandraTableHandle, partitions, clusteringKeyPredicates));
+    }
+
+    private static String extractClusteringKeyPredicates(CassandraPartitionResult partitionResult, CassandraTableHandle tableHandle, CassandraSession session)
```

Comment:
Looks similar to the logic in `io.prestosql.plugin.cassandra.CassandraMetadata#applyFilter`. Can we have a single method that does this?

Reply:
Tried that, but `CassandraClusteringPredicatesExtractor` is used to get also constraint in  `CassandraMetadata#applyFilter`, so use case is different